src/lib/ThumbnailsSidebar.js

```diff
@@ -314,8 +310,11 @@ class ThumbnailsSidebar {
      */
     setCurrentPage(pageNumber) {
         const parsedPageNumber = parseInt(pageNumber, 10);
-
-        if (parsedPageNumber >= 1 && parsedPageNumber <= this.pdfViewer.pagesCount) {
+        // don't use the virtual scroller if the pdfViewer is not ready but the preloader has opened the thumbnails
+        if (parsedPageNumber >= 1 && this.preloader?.thumbnailsOpen && this.pdfViewer.pagesCount === 0) {
+            this.currentPage = parsedPageNumber;
+            this.applyCurrentPageSelection();
+        } else if (parsedPageNumber >= 1 && parsedPageNumber <= this.pdfViewer.pagesCount) {
```

Comment:
Should we use `this.pdfViewer?.pageCount` to be safe?

Reply:
I actually left that part alone. If it gets there and pdfViewer is undefined then it should throw an error.